iceoryx_hoofs/design/include/iox/polymorphic_handler.hpp

```diff
@@ -14,18 +14,16 @@
 //
 // SPDX-License-Identifier: Apache-2.0
 
-#ifndef IOX_HOOFS_DESIGN_PATTERN_POLYMORPHIC_HANDLER_HPP
-#define IOX_HOOFS_DESIGN_PATTERN_POLYMORPHIC_HANDLER_HPP
+#ifndef IOX_HOOFS_DESIGN_POLYMORPHIC_HANDLER_HPP
+#define IOX_HOOFS_DESIGN_POLYMORPHIC_HANDLER_HPP
 
 #include <atomic>
 #include <type_traits>
 
-#include "iceoryx_hoofs/design_pattern/static_lifetime_guard.hpp"
+#include "iox/static_lifetime_guard.hpp"
 
 namespace iox
 {
-namespace design_pattern
-{
 namespace detail
```

Comment:
~~If the `PolymorphicHandler` is in `detail` shouldn't it also be in the folder `/detail`?~~ The namespace ends before the class definition start.s

Reply:
Oh, and could you please add this class to the overview `Readme.md`?